mmv1/templates/terraform/pre_create/alloydb_instance.go.erb

```diff
@@ -0,0 +1,18 @@
+// Read the config and call createsecondary api if instance_type is SECONDARY
+
+var instanceType interface{}
+var cluster interface{}
+
+if val, ok := obj["instanceType"]; ok {
+    instanceType = val
+}
+
+cluster = d.Get("cluster")
+
+if instanceType == "SECONDARY" {
+    if cluster != nil {
```

Comment:
Cluster's a required field, we should be fine. If we do want to confirm it exists anyways, we should use `GetOk` rather than `Get`.

Reply:
Removed the validation check for cluster as it is a required field